### DISCUSSION TITLE
TZIP-10 (Beacon) support for interacting with Tezos dApps

### DISCUSSION THREAD

**Discussion by rmlearney** [trustwallet/wallet-core#2124] [Ideas & Feature request]
WalletConnect is an elegant solution for linking mobile wallet to dApp. 

It seems Tezos have their own TZIP-10 standard for interacting with Tezos dApps, also called 'Beacon' (https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-10/tzip-10.md)

Would it be possible to incorporate Beacon support within Trustwallet (SDK at https://docs.walletbeacon.io)? The alternative is to get another Tezos-specific mobile app just to handle my single Tezos NFT.

I've posted the same issue in the upstream WalletConnect monorepo.

**Comment by hewigovens** [comment]
Is it this similar to Ethereum web3 provider?

**Comment by hewigovens** [comment]
Interesting, I would suggest WalletConnect V2 to add Tezos support

**Comment by rmlearney** [comment]
@hewigovens what's the best way to recommend this to them?

**Reply by hewigovens** [reply]
Tezos is added here already: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-26.md, maybe you can submit a feature request issue to https://github.com/WalletConnect/walletconnect-monorepo/issues

**Comment by rmlearney** [comment]
Hi @hewigovens you'll see I previously opened issue #900 in that repo. Are you able to add any comment there so they might notice it?
